astropy/table/table.py

```diff
@@ -2671,6 +2671,8 @@ def remove_columns(self, names):
         '''
         if isinstance(names, str):
             names = [names]
+        elif isinstance(names, types.GeneratorType):
```

Comment:
How about being even more and just do 
```
else:
    names = list(names)
```

Reply:
General interest question, do tuples still consume less memory? I know the speed differences are now resolved...

Reply:
I picked `list` mostly because the docstring talks about lists. I think for large numbers of elements, the difference in memory use is tiny - lists may have some extra allocation already from the start, while tuples obviously do not need it.